contracts/receivables/BaseCreditPoolReceivable.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import "hardhat/console.sol";
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "../BaseCreditPool.sol";
+
+contract BaseCreditPoolReceivable is ERC721, ERC721Enumerable, ERC721URIStorage, AccessControl {
+    using Counters for Counters.Counter;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    Counters.Counter private _tokenIdCounter;
+
+    mapping(uint256 => ReceivableInfo) public receivableInfoMapping;
+
+    struct ReceivableInfo {
+        BaseCreditPool baseCreditPool;
+        address receivableAsset;
```

Comment:
receivableAsset is always this contract. It seems unnecessary.

Reply:
I will rename this, receivableAsset is actually the ERC20 token this receivable is paid out in.